### ISSUE TITLE
evaluation on custom dataset (YoloV8 OBB)

### ISSUE THREAD

**Issue by Petros626** [ultralytics/ultralytics#9123]
### Search before asking

- [X] I have searched the YOLOv8 [issues](https://github.com/ultralytics/ultralytics/issues) and [discussions](https://github.com/ultralytics/ultralytics/discussions) and found no similar questions.

### Question

Is it possible to evaluate model  with the oriented bounding box on a custom dataset or only on DOTA? I read the documenation here: https://docs.ultralytics.com/de/tasks/obb/#val and didn't found a part about custom evaluation.

Thanks in advance

**Comment by glenn-jocher** [comment]
@Petros626 hello! 😊 Yes, you can absolutely evaluate models with oriented bounding boxes (OBBs) on custom datasets, not just DOTA. For this, you'll first need to prepare your custom dataset in a similar format as supported by YOLOv8 for OBB tasks. 

Make sure your dataset annotations are structured properly (refer to the DOTA or similar dataset formatting guidelines). After that, you can simply specify your custom dataset in the YAML file and use it for validation just like you would with any YOLO model.

Here's a quick example on how you might specify this in your custom dataset YAML and execute the validation process:

```yaml
# custom_dataset.yaml
path: ../datasets/my_custom_dataset  # dataset root dir
train: images/train  # train images relative path
val: images/val  # val images relative path
test:  # optional test images relative path

# Assuming 'classes.txt' lists all your classes
nc: 10  # number of classes
names: ['class1', 'class2', ..., 'class10']  # class names
```

And then run validation with:

```bash
yolo val data=custom_dataset.yaml model=yolov8n-obb.pt
```

Ensure your classes and paths are correctly defined according to your actual dataset structure. Happy evaluating! If you have more specific questions or run into issues, feel free to ask.

**Comment by Petros626** [comment]
@glenn-jocher thank you for the exact and fast answer!

**Comment by glenn-jocher** [comment]
@Petros626 You're welcome! 😊 If you have any more questions or need further assistance, don't hesitate to reach out. Happy detecting!

**Comment by MayPhyuKhin** [comment]
Hello  Ultralytics team.
I want to create a label dataset according to bounding box values and angles for Yolov8-obb training. How can I extract the labels?

**Comment by Petros626** [comment]
> Hello Ultralytics team. I want to create a label dataset according to bounding box values and angles for Yolov8-obb training. How can I extract the labels?

You should open a separate issue or search for a similar than using a totally different question.

**Comment by glenn-jocher** [comment]
Hello @Petros626,

Thank you for your question! To create a label dataset for YOLOv8-OBB training, you'll need to format your annotations in the YOLO OBB format, which specifies bounding boxes by their four corner points with coordinates normalized between 0 and 1. Here's a quick guide:

1. **Annotation Format**: Each line in your label file should follow this format:
   ```
   class_index, x1, y1, x2, y2, x3, y3, x4, y4
   ```
   For example:
   ```
   0 0.780811 0.743961 0.782371 0.74686 0.777691 0.752174 0.776131 0.749758
   ```

2. **Dataset Structure**: Ensure your dataset directory is structured properly with images and corresponding label files.

3. **YAML Configuration**: Create a YAML file to specify the paths and class names for your dataset. Here’s an example:
   ```yaml
   path: ../datasets/my_custom_dataset
   train: images/train
   val: images/val
   nc: 10
   names: ['class1', 'class2', ..., 'class10']
   ```

4. **Conversion Script**: If your current annotations are in a different format (e.g., DOTA), you can use the provided conversion script:
   ```python
   from ultralytics.data.converter import convert_dota_to_yolo_obb
   convert_dota_to_yolo_obb("path/to/DOTA")
   ```

5. **Training**: Once your dataset is ready, you can start training:
   ```bash
   yolo detect train data=custom_dataset.yaml model=yolov8n-obb.pt epochs=100 imgsz=640
   ```

For more detailed instructions, please refer to our [documentation](https://docs.ultralytics.com/datasets/obb/).

If you encounter any issues, please provide a reproducible example to help us assist you better: [Minimum Reproducible Example](https://docs.ultralytics.com/help/minimum_reproducible_example).

Best of luck with your training! 🚀
